lib/aggregate.js

```diff
@@ -963,7 +963,7 @@ Aggregate.prototype.pipeline = function() {
  *     const result = await aggregate.exec();
  *
  * @param {Function} [callback]
- * @return {Promise} Returns a Promise if no "callback" is given.
+ * @return {Promise<Document[]>} Returns a Promise if no "callback" is given.
```

Comment:
I don't think that's correct.

Aggregate execution results:
1- Plain JS objects, mongoose doesn't do any parsing on them, so they do not extend the `Document` class.
2- Can be of any type, for example, you can `$addFields` or `$group`, which changes the shape of the return type from the defined schema to anything, that's why it's a design decision for mongoose not to do anything for aggregation results.

Reply:
thanks for explaining, i will close this PR then, i had just guessed from some tests and the surrounding code (i dont know much about mongodb aggregate)